rfc-175-frontend-fewer-apps.md

```diff
@@ -0,0 +1,260 @@
+* `frontend` has a one-controller per document type layout, which is preferable for document types with multiple pages as it follows [rails conventions for controllers] to do the orchestration.
+
+[government-frontend] was also considered for the host app, but not chosen because it has a more restrictive presenter pattern that works well for single-page document types. Even though `government-frontend` has many document types and accounts for 50% of the frontend traffic, it would be easier to move the single-page document types from `government-frontend` to `frontend` than it would to try and force the document types in `frontend` into the presenter pattern.
+
+### Testing standardisation
+We propose standardising around RSpec as the test suite for the frontend, ideally [modernising the tests], for example by switching to request tests rather than controller tests and system tests rather than feature tests. RSpec is the [preferred testing suite] on GOV.UK and is favoured by many developers because the tests and testing output are easier to read. Of the 4 applications only [government-frontend] uses minitest at this point, so its test suite could be converted route-by-route as it is merged into [frontend]
+
+### Code structure
+The code structure will be standardised. As well as using the [default Rails directory structure], we propose the following standard for additional classes (e.g. services, api calls). The standard expands on the [conventions] documented in the developer docs.
+
+The host app will follow the default MVC structure of Rails applications, with some additional restrictions on what goes where that are frontend specific (listed below)
+
+```
+├── app
+│   ├── assets
+│   ├── controllers
+│   ├── helpers
+│   ├── models
+│   ├── presenters
+│   ├── views
+├── config
+│   ├── data
+│   ├── locales
+├── data
+├── lib
+│   ├── tasks
+└── .gitignore
+```
+
+Under `/app`:
+
+`helpers`
+- General utility methods for a view.
+
+`models`
+- Frontend apps do not have database records, so models here are created to store data received from api calls (eg content items from content store), process data that will be written to apis (eg feedback items to zendesk), and information ingested from YAML/JSON/CSVs.
+
+`presenters`
+- Presentation related transformations on model data. It is a wrapper around a model for use in a view.
+
+Under `/lib`
+
+`tasks`
+- Rake tasks
+
+Use `lib` rather than `app/lib` for custom code that enhances the app but doesn’t fit into a directory in `app`
+
+Under `/config`
+
+`data`
+- YAML, JSON, CSV config files. Data files should be placed here if they are clearly used in the configuration of the app. Ideally, though, we should **avoid using data files** for configuration where possible (see `/data` below)
+
+`locales`
+- Static content like lists of contact information should ideally be stored in the locale files to promote localisation.
+
+Under `/data`
```

Comment:
Nitpicking further but this seems less conventional than using `lib/data` since you'll be adding a new top level directory, whereas lib is already intended to be expanded

Reply:
Yeah sounds good about it being a moot point - there's something smelly about a directory named data, reminds me of other dubious ones like "shared" and "util".

I think the main thing is to avoid there being two `data` directories - one under `/config/data` and another one as this introduces further ambiguity. 

I do agree that there is not a formally defined location in Rails, so does veer into the realm of individual interpretations of the enigmatic "Rails way". Would also agree that `app/data` is the most surprising suggestion so far. I wouldn't share your concern that it's weird something in `app` referencing a resource from `lib` but I know that veers into the other great Rails debate about the relationship of those directories.